system-tests/src/test/java/org/eclipse/edc/samples/transfer/Transfer04openTelemetryTest.java

```diff
@@ -0,0 +1,85 @@
+/*
```

Comment:
I see a few of issues with this test:

1. Code should not download remote files like this. Gradle should handle all dependencies
2. The Open Telemetry agent is not needed unless it is explicitly being tested, which it should not be in a sample
3. Overall, I don't understand the purpose of this test. What is it testing?

Reply:
Hi Jim, I picked up on this PR 
The only thing i added was the file download of the opentelemetry jar. I can move this to the Gradle script ofc.
I assumed the PR was close to be ready-to-merge as in the original PR it was mentioned that the failing tests is the only thing missing. We could ofc rethink the whole solution.

I think we could test the basic workflow like with the previous samples and test if telemetry logs are written or if Jaeger is reachable.

Reply:
I agree that the agent should be downloaded by gradle. The test is needed because we decided to add E2E tests on all the samples to avoid regressions on edc upgrades (testing samples manually is a tedious work). This test was provided time ago, I think it still makes sense.

Reply:
OK, but two points:

1. Is it testing that the agent receives messages? 
2. Open telemetry should be tested in the main repository, tests here should focus on the bootstrap of the samples. In other words, we should be confident open telemetry works with EDC based on the main repo test coverage, and here we only test the functionality added by the sample.

Reply:
The idea was to have the tests follow the sample steps from start to end. This should not test the interaction with open telemetry but that what's written in the README file is still working

Reply:
OK so then we can eliminate the open telemetry agent download entirely since it is not under test. The test should just verify the end-to-end flow.

Reply:
The agent is part of the docker compose file which we use to initialize the testcontainer
We'd have to exclude the agent from the docker compose file for the test case but I wouldn't do so as:

- We'll have two different docker compose files
- We're not testing the original sample setup anymore

I don't think it hurts to just leave the agent download there

Reply:
OK sounds good, but then we should be testing the same configuration/image that is used in the sample. That would mean not downloading the agent in the test and using the runtime setup from the Gradle project build.

Reply:
That's what we do now
If you check the gradle builds files you'll see that the agent download is a part of the gradle build step

Reply:
Can you please use Gradle's dependency resolution mechanism a pull the dependency from the local cache when building?

https://mvnrepository.com/artifact/io.opentelemetry.javaagent/opentelemetry-javaagent

This will allow us to track dependencies, use Gradle settings for downloads, and enable caching.